src/coreclr/vm/object.cpp

```diff
@@ -1545,13 +1558,11 @@ void StackTraceArray::CopyFrom(StackTraceArray const & src)
     }
     CONTRACTL_END;
 
-    m_array = (I1ARRAYREF) AllocatePrimitiveArray(ELEMENT_TYPE_I1, static_cast<DWORD>(src.Capacity()));
-
-    Volatile<size_t> size = src.Size();
+    Volatile<uint32_t> size = src.Size();
```

Comment:
```suggestion
    uint32_t size = src.Size();
```
Local `Volatile<T>` variable does not make sense. If the Size() reading needs to have read memory barrier, the barrier should be on the reading side.

Reply:
It's been there from the first git commit to the repo. I was trying to reason about whether the memory barrier it introduces is needed too and came to the same conclusion as you.

Reply:
Changed